### PR TITLE
ENH: Add an example that demonstrates interfacing with NumPy

### PR DESCRIPTION
**Pull request by andinet**

### CODE REVIEW

File: src/Bridge/CMakeLists.txt
Comment (tbirdso) [inline src/Bridge/CMakeLists.txt:5]: Replace with `add_subdirectory_if_module_enabled` so that this directory is skipped if the user has disabled NumPy support in their ITK build. See [https://github.com/InsightSoftwareConsortium/ITKSphinxExamples/blob/master/src/Core/CMakeLists.txt#L1](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples/blob/master/src/Core/CMakeLists.txt#L1) for example usage

**Review by tbirdso — CHANGES_REQUESTED** [review]
@andinet The example is looking great! I have requested a couple of CMake changes below.

**Comment by andinet** [comment]
@tbirdso please take a look. I addressed some of your comments. 

I still need to update the notebook

**Review by tbirdso — COMMENTED** [review]
Thank you for the fixes @andinet , the writeups you added look great. Will wait on notebook fixes for final approval.

**Comment by andinet** [comment]
@tbirdso 

Any thoughts on why the Lint and Documentation tests are failing?

**Comment by tbirdso** [comment]
> 2022-05-20T16:30:24.4633969Z 
2022-05-20T16:30:24.4634082Z Notebook error:
2022-05-20T16:30:24.4634518Z NoSuchKernel in src/Bridge/NumPy/ConvertNumPyArrayToitkImage/ConvertNumPyArrayToitkImage.ipynb:
2022-05-20T16:30:24.4634961Z No such kernel named python3
2022-05-20T16:30:24.4635272Z ninja: build stopped: subcommand failed.
2022-05-20T16:30:24.4635598Z ninja: build stopped: subcommand failed.
2022-05-20T16:30:24.4635908Z Command exited with the value: 1
2022-05-20T16:30:24.4642646Z Error(s) when building project
2022-05-20T16:30:24.4643474Z MakeCommand:/home/runner/work/_temp/685782585/cmake-3.18.3-Linux-x86_64/bin/cmake --build . --config "Release"
2022-05-20T16:30:24.4644323Z    0 Compiler errors
2022-05-20T16:30:24.4649303Z    1 Compiler warnings
2022-05-20T16:30:24.4649940Z SetCTestConfiguration:BuildDirectory:/home/runner/work/bld/ITKEx-build
2022-05-20T16:30:24.4650387Z SetCTestConfiguration:SourceDirectory:/home/runner/work/Ex

File: src/Bridge/NumPy/ConvertNumPyArrayToitkImage/CMakeLists.txt
Comment (tbirdso) [inline src/Bridge/NumPy/ConvertNumPyArrayToitkImage/CMakeLists.txt:14]: These files need to be prefixed with `${CMAKE_CURRENT_BINARY_DIR}/` so that CTest gets the absolute path to each image file.

**Review by tbirdso — CHANGES_REQUESTED** [review]

**Comment by andinet** [comment]
@tbirdso @thewtex 

The one ctest error/warning is still there despite several attempts.  I have run out of ideas

**Comment by thewtex** [comment]
@andinet awesome work, here! :1st_place_medal: :sparkles: :heart: 

I was able to reproduce a documentation build error locally -- working on a patch.

**Review by thewtex — APPROVED** [review]
:sparkler:

**Comment by tbirdso** [comment]
💯 Thank you for your persistence on this one @andinet and @thewtex !

**Pull request merged**
